common/src/main/java/com/wynntils/models/players/profile/GuildProfile.java

```diff
@@ -0,0 +1,43 @@
+/*
+ * Copyright © Wynntils 2023.
+ * This file is released under LGPLv3. See LICENSE for full license details.
+ */
+package com.wynntils.models.players.profile;
+
+import com.google.gson.JsonDeserializationContext;
+import com.google.gson.JsonDeserializer;
+import com.google.gson.JsonElement;
+import com.google.gson.JsonObject;
+import com.google.gson.JsonParseException;
+import com.wynntils.utils.colors.CustomColor;
+import java.lang.reflect.Type;
+
+public record GuildProfile(String prefix, String name, CustomColor color) {
+    public static class GuildProfileDeserializer implements JsonDeserializer<GuildProfile> {
+        @Override
+        public GuildProfile deserialize(JsonElement jsonElement, Type type, JsonDeserializationContext context)
+                throws JsonParseException {
+            JsonObject jsonObject = jsonElement.getAsJsonObject();
+
+            String name = jsonObject.getAsJsonPrimitive("_id").getAsString();
+            String prefix = jsonObject.getAsJsonPrimitive("prefix").getAsString();
+
+            // Color can be null, "", or a hex string
+            String colorString = jsonObject.has("color")
+                    ? jsonObject.getAsJsonPrimitive("color").getAsString()
+                    : "";
+
+            // If color is null or empty, use the name to generate a color
+            CustomColor color = !colorString.isBlank()
+                    ? CustomColor.fromHexString(colorString)
```

Comment:
Could this just be `CustomColor.fromHexString` as if it doesn't match the hex pattern it'll return CustomColor.NONE and the edge case will catch it anyway

Reply:
Could be, maybe it just doesn't highligh the edge cases as well. Im fine with it either way